changelogutils/changelogdocutils/merged_release.go

```diff
@@ -0,0 +1,201 @@
+package changelogdocutils
+
+import (
+    "context"
+    "encoding/json"
+    "fmt"
+    "github.com/google/go-github/v32/github"
+    "github.com/solo-io/go-utils/githubutils"
+    . "github.com/solo-io/go-utils/versionutils"
+    "regexp"
+)
+
+type DependencyFn func(*Version) (*Version, error)
+
+type Options struct {
+    // Number of versions from the most previous version to display in changelog
+    NumVersions int
+    // Maximum version changelog to display on this changelog
+    MaxVersion *Version
+    // Minimum version changelog to display on this changelog
+    MinVersion *Version
+    ProjectName,
+    RepoOwner,
+    MainRepo,
+    DependentRepo string
+}
+
+type MergedReleaseGenerator struct {
+    client         *github.Client
+    releaseDepMap  map[Version]*Version
+    opts           Options
+    DependencyFunc DependencyFn
+}
+
+func NewMergedReleaseGenerator(opts Options, client *github.Client) *MergedReleaseGenerator {
+    return &MergedReleaseGenerator{
+        opts:          opts,
+        client:        client,
+        releaseDepMap: map[Version]*Version{},
+    }
+}
+
+func NewMergedReleaseGeneratorWithDepFn(opts Options, client *github.Client, depFn DependencyFn) *MergedReleaseGenerator {
+	gen :=  NewMergedReleaseGenerator(opts, client)
+	gen.DependencyFunc = depFn
+	return gen
+}
+
+/*
+ The merged release generator has 3 steps:
+ 1. Fetches enterprise repo release notes
+ 2. Gets open source dependency for each enterprise version
+ 3. Merges open source release notes into enterprise version release notes
+*/
+func (g *MergedReleaseGenerator) GenerateJSON(ctx context.Context) (string, error) {
+    var err error
+    enterpriseReleases, err := g.GetMergedEnterpriseRelease(ctx)
+    if err != nil {
+        return "", err
+    }
+    var out struct {
+        Opts        Options
+        ReleaseData *ReleaseData
+    }
+    out.Opts = g.opts
+    out.ReleaseData = enterpriseReleases
+    res, err := json.Marshal(out)
+    return string(res), err
+}
+
+func (g *MergedReleaseGenerator) GetMergedEnterpriseRelease(ctx context.Context) (*ReleaseData, error) {
+    ossReleases, err := NewMinorReleaseGroupedChangelogGenerator(Options{
+        RepoOwner: "solo-io",
```

Comment:
`solo-io` shouldn't be hardcoded here, fixed. 

As far as specifying both repos, this will cause a few extra if statements and checks that I'm okay with forgoing for the sake of simplicity. Especially since we're probably the only ones using this.